src/astgen.zig

```diff
@@ -2221,6 +2221,20 @@ fn import(mod: *Module, scope: *Scope, call: *ast.Node.BuiltinCall) InnerError!*
     return addZIRUnOp(mod, scope, src, .import, target);
 }
 
+fn typeOf(mod: *Module, scope: *Scope, rl: ResultLoc, call: *ast.Node.BuiltinCall) InnerError!*zir.Inst {
+    const tree = scope.tree();
+    const arena = scope.arena();
+    const src = tree.token_locs[call.builtin_token].start;
+    const params = call.params();
+    if (params.len < 1) {
+        return mod.failTok(scope, call.builtin_token, "expected at least 1 argument, found 0", .{});
+    }
+    var items = try arena.alloc(*zir.Inst, params.len);
```

Comment:
if `params.len == 1` use the `typeof` instruction since those are much more common than peer resolution `@TypeOf`s.